packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1054,9 +1104,39 @@ class EditableText extends StatefulWidget {
   /// {@template flutter.widgets.editableText.inputFormatters}
   /// Optional input validation and formatting overrides.
   ///
-  /// Formatters are run in the provided order when the text input changes. When
-  /// this parameter changes, the new formatters will not be applied until the
-  /// next time the user inserts or deletes text.
+  /// Formatters are run in the provided order when the user changes the text
+  /// contained in the widget. They're not applied when the changes are
+  /// selection only, or not initiated by the user.
+  ///
+  /// When this widget rebuilds, each input formatter in the new widget's
+  /// [inputFormatters] list checks the configuration of the input formatter
+  /// from the same location in the old [inputFormatters], to determine if the
+  /// new formatters need to be re-applied to the current [TextEditingValue] of
+  /// this widget.
+  ///
+  /// {@tool snippet}
+  ///
+  /// The following code uses a combination of 2 [TextInputFormatter]s and a
+  /// fictional `UsPhoneNumberFormatter` (which simply adds parentheses and
+  /// hypens), to turn user input into a valid United States telephone number,
+  /// in the format of (123)456-7890.
+  ///
+  /// The combined effect of the 3 formatters is idempotent, meaning applying it
+  /// to an already formatted value is a no-op. This is a required property for
+  /// user-supplied formatters. The fictional `UsPhoneNumberFormatter` is not
```

Comment:
NIt:

"This is a required property for user-supplied formatters."

=>

"This is a requirement for user-supplied formatters."

I thought that saying "property" was confusing because it made me think of a property of a class or something.